fetchDependencies

```diff
@@ -444,6 +446,15 @@ EXT_DEPS=ExternalDependencies
 XC_PROJ="${EXT_DEPS}.xcodeproj"
 XC_DD_PATH="${EXT_DIR}/build"
 
+# Determine if xcpretty is present
+XCPRETTY_PATH=$(command -v xcpretty 2> /dev/null)
+
+# If xcpretty is present, use it to format xcodebuild output
+XC_OUTPUT_FMT_CMD="tee"
```

Comment:
You had the main build write output to a log, but not `fetchDependencies`. Why?

Reply:
Handling the various combinations of options in `fetchDependencies` is more problematic (such as parallel builds), and I was trying to minimize changes in a single PR.

That said, I have refactored `fetchDependencies` to handle this better for the "common" case - it'll now use `tee` to output the full dependency build output to a log when using `xcpretty`, and use neither `xcpretty` nor `tee` if `--parallel-build` is specified.